lib/rules/no-tabs.js

```diff
@@ -21,25 +21,54 @@ module.exports = {
             category: "Stylistic Issues",
             recommended: false
         },
-        schema: []
+        schema: [],
+        fixable: "whitespace"
     },
 
     create(context) {
         return {
             Program(node) {
+                const source = context.getSourceCode();
+                let scriptIndex = 0;
+
                 context.getSourceLines().forEach((line, index) => {
-                    const match = regex.exec(line);
+                    let match;
+
+                    // .exec() returns null when it cannot find more matches
+                    while ((match = regex.exec(line))) {
+                        const range = [
+                            scriptIndex + match.index,
+                            scriptIndex + match.index + 1
+                        ];
+                        const astNode = source.getNodeByRangeIndex(range[0]);
 
-                    if (match) {
-                        context.report(
+                        context.report({
                             node,
-                            {
+                            loc: {
                                 line: index + 1,
                                 column: match.index + 1
                             },
-                            "Unexpected tab character."
-                        );
+                            message: "Unexpected tab character.",
+
+                            // This function needs to capture the current `range`
+                            // eslint-disable-next-line no-loop-func
+                            fix(fixer) {
+
+                                // Fixing string/template/regex literals is left to the developer
+                                // because it could change the semantics of the code
+                                if (
+                                    astNode.type === "Literal" ||
+                                    astNode.type === "TemplateElement") {
+                                    return null;
+                                }
+
+                                return fixer.replaceTextRange(range, " ");
+                            }
+                        });
                     }
+
+                    // The + 1 is to account for linebreaks
+                    scriptIndex += line.length + 1;
```

Comment:
Interesting! I must say I am not really sure how to tackle this. In theory, a single file could use both types of line breaks, so I would have to somehow consult the actual source (from `context.getSourceCode().text`) and determine the line break type for current line... How would you do this? Or, do you know of any other rule which would use a similar mechanism that I can get some inspiration from? Or, even better, is there a better way to determine the current range for the detected tab character?